src/main/java/org/opentripplanner/transit/raptor/api/path/Path.java

```diff
@@ -328,8 +329,19 @@ public int compareTo(Path<T> other) {
         AccessPathLeg<S> accessLeg, EgressPathLeg<S> egressPathLeg
     ) {
         return accessLeg.access().numberOfRides()
-            // Skip first transit
-            + (int)accessLeg.nextLeg().nextLeg().stream().filter(PathLeg::isTransitLeg).count()
-            + egressPathLeg.egress().numberOfRides();
+            + (int) accessLeg
+                .nextLeg()
+                .stream()
+                .filter(PathLeg::isTransitLeg)
+                // Filter out legs, where the user stays onboard to the next leg
+                .filter(leg -> {
+                    var transfer = leg.asTransitLeg().getConstrainedTransferAfterLeg();
+                    return transfer == null || !transfer.getTransferConstraint().isStaySeated();
+                })
+                .count()
+            + egressPathLeg.egress().numberOfRides()
+            // Remove one boarding to get the count of transfers only.
+            // We can't remove the first leg, as it might have info on a stay seated transfer.
+            - 1;
```

Comment:
// Something like this would make this easier to read:
```java
        int nAccessRides = accessLeg.access().numberOfRides();
        int nTransitRides = (int) accessLeg
                .nextLeg()
                .stream()
                .filter(PathLeg::isTransitLeg)
                // Filter out legs, where the user stays onboard to the next leg
                .filter(leg -> !leg.asTransitLeg().isStaySeatedOnToNextLeg())
                .count();
        int nEgressRides = egressPathLeg.egress().numberOfRides();

        // Remove one boarding to get the count of transfers only.
        return nAccessRides + nTransitRides + nEgressRides - 1;
```
And in TransitPathLeg
```java
    public boolean isStaySeatedOnToNextLeg() {
        return constrainedTransferAfterLeg != null && 
               constrainedTransferAfterLeg.getTransferConstraint().isStaySeated();
    }
```

Reply:
Yes, much easier to read this way.